packages/core/contracts/Escrow.sol

```diff
@@ -297,20 +307,27 @@ contract Escrow is IEscrow, ReentrancyGuard {
         SafeERC20.safeTransfer(IERC20(token), to, value);
     }
 
-    function _storeResult(string memory _url, string memory _hash) internal {
-        bool writeOnchain = bytes(_hash).length != 0 || bytes(_url).length != 0;
-        if (writeOnchain) {
-            // Be sure both of them are not zero
-            finalResultsUrl = _url;
-            finalResultsHash = _hash;
-        }
-    }
-
     modifier trusted() {
         require(areTrustedHandlers[msg.sender], 'Address calling not trusted');
         _;
     }
 
+    modifier trustedOrReputationOracle() {
+        require(
+            areTrustedHandlers[msg.sender] || msg.sender == reputationOracle,
+            'Address calling not trusted'
+        );
+        _;
+    }
+
+    modifier trustedOrRecordingOracle() {
```

Comment:
This modifier might not be worth since it is used only once. Check the weight of the contract, if it is lighter by removing it, we should consider to remove it